src/birthder.js

```diff
@@ -573,6 +580,81 @@
       msg.send(`*Birthdays list*\n${message.join('\n')}`)
     })
 
+    // Reset date of first working day.
+    robot.respond(routes.start, async (msg) => {
+      let date
+      let name
+      let user
+      let users
+
+      if (!await routines.isAdmin(robot, msg.message.user.name.toString())) {
+        msg.send(MSG_PERMISSION_DENIED)
+        return
+      }
+
+      name = msg.match[2].trim()
+      date = msg.match[3]
+      users = []
+
+      for (const u of robot.brain.usersForFuzzyName(name)) {
+        if (await routines.isUserActive(robot, u)) {
+          users.push(u)
+        }
+      }
+
+      if (!routines.isValidDate(date, DATE_FORMAT)) {
+        msg.send(MSG_INVALID_DATE)
+        return
+      }
+
+      if (users.length === 1) {
+        user = users[0]
+        user.dateOfStart = date
+
+        return msg.send(`Saving ${name}'s first working day.`)
+      } else if (users.length > 1) {
+        return msg.send(getAmbiguousUserText(users))
+      } else {
+        return msg.send(`I have never met ${name}.`)
+      }
+    })
+
+    // Print sorted users first working days.
+    robot.respond(routes.slist, async function (msg) {
+      let message
+
+      const allUsers = []
+      for (const u of Object.values(robot.brain.data.users)) {
+        if (await routines.isUserActive(robot, u)) {
+          allUsers.push(u)
+        }
+      }
+
+      message = allUsers
+        .filter(user => routines.isValidDate(user.dateOfStart, DATE_FORMAT))
+        .map(user => {
+          const thisYear = moment().year()
+          const date = moment(user.dateOfStart, DATE_FORMAT).year(thisYear)
+          const sortedDate = date.unix() >= moment().unix()
+            ? date.format(`DD.MM.${date.year() - 1}`) : date.format(`DD.MM.${date.year()}`)
+
+          return {
+            name: user.name,
+            dateOfStart: user.dateOfStart,
+            sortedDate: sortedDate
+          }
+        })
+        .sort((a, b) => sorting(a.sortedDate, b.sortedDate, OUTPUT_DATE_FORMAT))
+        .map(user => ` @${user.name} join our team ${moment(user.dateOfStart, DATE_FORMAT).format(OUTPUT_DATE_FORMAT)}`)
```

Comment:
You mostly copy the filtering from `birthday list` command, may be better will be not repeat but reuse this piece of code and strike it out as a function.